test/async.vimspec

```diff
@@ -0,0 +1,51 @@
+let s:output = []
+function! s:on_stdout(id, data, event) abort
+  call add(s:output, join(a:data, ''))
+  call async#job#stop(a:id)
+endfunction
+
+Describe async
+  Describe async#job#start
+    It can start cmmand and return numbered job-id
+      let job = async#job#start('vim --version', {})
+      Assert Equals(type(job), v:t_number)
+    End
+  End
+
+  Describe async#job#stop
+    It can stop the job specified
+      let job = async#job#start('bash -c "sleep 2 && touch i-love-vim"', {})
+      sleep 3
+      call async#job#stop(job)
+      Assert filereadable('i-love-vim')
+      call delete('i-love-vim')
+
+      let job = async#job#start('bash -c "sleep 2 && touch i-love-vim"', {})
+      sleep 1
+      call async#job#stop(job)
+      Assert !filereadable('i-love-vim')
+    End
+  End
+
+  Describe async#job#wait
+    It can wait the job specified
+      let job = async#job#start('bash -c "sleep 2"', {})
+      let start = reltime()
+      call async#job#wait([job])
+      let seconds = reltimefloat(reltime(start))
+      call async#job#stop(job)
+      Assert seconds > 2
+    End
+  End
+
+  Describe async#job#send
+    It can send input text to the job
+      let s:output = []
+      let job = async#job#start('cat', {'on_stdout': function('s:on_stdout')})
```

Comment:
Can we instead do this? That way we can avoid global var.

```vim

function! s:on_stdout(ctx, id, data, event) abort
    call add(a:ctx,output, join(a:data, ''))
endfunction

let ctx = { 'output': [] }
let job = async#job#start('cat', {'on_stdout': function('s:on_stdout', [ctx])})
```
If possible I would prefer lambdas and then use `job#wait` in the next line after job.

Reply:
I'm not familier to neovim but neovim already support parcial arguments?